cristin-import/src/main/java/no/unit/nva/cristin/lambda/FilenameEventEmitter.java

```diff
@@ -43,32 +52,59 @@ public FilenameEventEmitter(S3Client s3Client, EventBridgeClient eventBridgeClie
     @Override
     public void handleRequest(InputStream input, OutputStream output, Context context) throws IOException {
         ImportRequest importRequest = parseInput(input);
-        List<String> files = listFiles(importRequest);
+        List<URI> files = listFiles(importRequest);
         validateLocation(importRequest, files);
-
         List<PutEventsResult> failedRequests = emitEvents(context, files);
         logWarningForNotEmittedFilenames(failedRequests);
         returnNotEmittedFilenames(output, failedRequests);
     }
 
+    @JacocoGenerated
+    private static EventBridgeClient defaultEventBridgeClient() {
+        return EventBridgeClient.builder()
+                   .region(ApplicationConstants.AWS_REGION)
+                   .httpClient(UrlConnectionHttpClient.create())
+                   .build();
+    }
+
+    @JacocoGenerated
+    private static S3Client defaultS3Client() {
+        return S3Client.builder()
+                   .region(ApplicationConstants.AWS_REGION)
+                   .httpClient(UrlConnectionHttpClient.create())
+                   .build();
+    }
+
+    private URI createUri(URI s3Location, String filename) {
```

Comment:
The S3 client does not return URIs when one is listing files. The URI has to be recomposed. Any better way to create a URI where we do the following are welcome:

```
"https://someHost.org/partial/path" + "continuation/of/the/path" = "https://someHost.org/partial/path/continuation/of/the/path"

```